src/Entities/Address.php

```diff
@@ -1,250 +1,146 @@
 <?php
 
+declare(strict_types=1);
+
 namespace GingerPluginSdk\Entities;
 
-use GingerPluginSdk\Exceptions\OutOfEnumException;
-use GingerPluginSdk\Helpers\FieldsValidatorTrait;
 use GingerPluginSdk\Helpers\MultiFieldsEntityTrait;
+use GingerPluginSdk\Helpers\SingleFieldTrait;
 use GingerPluginSdk\Interfaces\MultiFieldsEntityInterface;
-use GingerPluginSdk\Interfaces\ValidateFieldsInterface;
 use GingerPluginSdk\Bases\BaseField;
+use GingerPluginSdk\Properties\Country;
 use JetBrains\PhpStorm\Pure;
 
-class Address implements MultiFieldsEntityInterface, ValidateFieldsInterface
+final class Address implements MultiFieldsEntityInterface
 {
-    use FieldsValidatorTrait;
-    use MultiFieldsEntityTrait {
-        toArray as defaultToArray;
-    }
+    use SingleFieldTrait;
+    use MultiFieldsEntityTrait;
 
-    private array $fields;
     private BaseField $address_type;
     private BaseField $postal_code;
-    private BaseField $address;
-    private BaseField $housenumber;
     private BaseField $country;
-    private BaseField|null $city = null;
-    private BaseField|null $street = null;
-    private array $acceptable_types;
-
-    #[Pure] public function __construct()
-    {
-        $this->fields = [
-            'address_type' => true,
-            'address' => true,
-            'postal_code' => true,
-            'housenumber' => true,
-            'country' => true,
-        ];
-    }
+    private BaseField $city;
```

Comment:
The BaseField type appears to make the code more abstract and indirect. Personally I would prefer using simple scalar values here. If there's any validation that you want to perform this can be done in the constructor or using a value object.

Reply:
Basically, the big opportunity that comes with `BaseField` is to make `toArray` method and other methods that required work only with API entity fields, more automated and scaling intuitive.